syntax = "proto3";

package protos;
option go_package = "orch-go/gen/protos;protos";
option csharp_namespace = "Core";

import "custom_types.proto";
import "google/protobuf/empty.proto";
import "deposit.proto";

service DepositAnalyticService {
  rpc ProcessGetAll (GetAllRequest) returns (GetAllDepositsResponse);
  rpc ProcessGetById (GetDepositByIdRequest) returns (DepositModel);
  rpc ProcessGetByIds (GetDepositByIdsRequest) returns (GetAllDepositsResponse);
  rpc ProcessGetByCustomer (GetDepositsByCustomerRequest) returns (GetAllDepositsResponse);
  rpc ProcessGetCount (google.protobuf.Empty) returns (CountResponse);
  rpc ProcessGetCountByStatus (GetDepositCountByStatusRequest) returns (CountResponse);
}
